src/spatial/detail/ArborX_IndexableGetter.hpp

```diff
@@ -68,6 +76,16 @@ struct Indexables
   KOKKOS_FUNCTION auto size() const { return _values.size(); }
 };
 
-} // namespace ArborX::Details
+template <typename Values, typename IndexableGetter>
+#if KOKKOS_VERSION >= 40400
+KOKKOS_DEDUCTION_GUIDE
+#else
+KOKKOS_FUNCTION
+#endif
+    Indexables(Values, IndexableGetter) -> Indexables<Values, IndexableGetter>;
```

Comment:
Please remind me why we need to provide a user-defined deduction guide and it does not work out of the box.

Reply:
Implicitly-generated deduction guide for aggregates is C++20.